client/client.go

```diff
@@ -249,95 +279,63 @@ func (client *Client) err(e error) {
 	}
 }
 
-func (client *Client) handleResponse(key string, resp *Response) *Response {
-	if h, ok := client.respHandler.get(key); ok {
-		h(resp)
-		return nil
-	}
-	return resp
+func (client *Client) request() *request {
+	return client.requestPool.Get().(*request)
 }
 
-func (client *Client) handleInner(key string, resp *Response) *Response {
-	if h, ok := client.innerHandler.get(key); ok {
-		h(resp)
-		client.innerHandler.remove(key)
-		return nil
-	}
-	return resp
-}
+func (client *Client) submit(pt rt.PT, funcname string, arg []byte) (string, error) {
```

Comment:
you could also define `var err error` in the return type, right? What's best practice there?